src/domainobjectfactories/depot_position_factory.py

```diff
@@ -27,57 +27,106 @@ def create(self, record_count, start_id):
         dict
             A single depot position object
         """
 
-        position_type = self.create_position_type()
-        knowledge_date = self.create_knowledge_date()
+        isin, cusip, market = self.__create_instrument_details()
+
         record = {
-            'isin': instrument['isin'],
-            'knowledge_date': knowledge_date,
-            'position_type': position_type,
-            'effective_date': self.create_effective_date(
-                                2, knowledge_date, position_type),
-            'account': self.create_account(),
-            'qty': self.create_random_integer(),
-            'purpose': self.create_purpose(),
-            'depot_id': self.create_random_integer(length=5),
-            'time_stamp': datetime.now()
+            'as_of_date': self.__create_as_of_date(),
+            'value_date': self.__create_value_date(),
+            'isin': isin,
+            'cusip': cusip,
+            'market': market,
+            'depot_id': self.__create_depot_id(),
+            'purpose': self.__create_purpose(),
+            'quantity': self.__create_quantity()
         }
 
         for key, value in self.create_dummy_field_generator():
             record[key] = value
 
         return record
 
-    def create_purpose(self):
+    @staticmethod
+    def __create_as_of_date():
+        """ Return the 'as of date', which must be the current date
+        Returns
+        -------
+        Date
+            Date object representing the current date
+        """
+        return datetime.date.today()
+
+    @staticmethod
+    def __create_value_date():
+        """ Return the 'value date', which must be today or in 2 days time
+        Returns
+        -------
+        Date
+            Date object representing the current date or the date in 2 days
+            time
+        """
+        today = datetime.date.today()
+        day_after_tomorrow = today + datetime.timedelta(days=2)
+        return random.choice((today, day_after_tomorrow))
+
+    def __create_instrument_details(self):
+        """ Return the isin, cusip and market of an instrument persisted in the
+        local database.
+
+        Returns
+        -------
+        String
+            isin of instrument from local database
+        String
+            cusip of instrument from local database
+        String
+            market of instrument from local database
+        """
+        instrument = self.get_random_instrument()
+        isin = instrument['isin']
+        cusip = instrument['cusip']
+        market = instrument['market']
+        return isin, cusip, market
+
+    def __create_depot_id(self):
+        """ Return the account id value of an account persisted in the
+        database that is type 'Depot'
+
+        Returns
+        -------
+        String
+            account id of 'Depot' type account from database
+        """
+        account = self.get_random_account()
+        while account['account_type'] != 'Depot':
```

Comment:
This feels like similar logic to FrontOfficePositionFactory.__create_account_id (and maybe others elsewhere), where we keep selecting records until we find one where one of the fields doesn't match a specific value.
I think you can refactor this out so that the methods in the factories call a new method which looks like:
getRandomRecordWhereFieldDoesNotMatch(tableName, fieldToCheck, collectionOfValuesTheFieldMustNotMatch)
As at the moment we've got a few places where we've got this logic that could in theory loop forever